plugin.js

```diff
@@ -1,167 +1,204 @@
+            )
+          ),
+          cssModule: path.join("plugins", plugin.id, cssModule),
+          jsModule: path.join("plugins", plugin.id, jsModule),
+        };
+
+        fs.writeFileSync(
+          path.join(Config.build_path(exp_info.src), plugin.target),
+          hb.compile(page_template.toString())(page_data)
+        );
+
+        plugins.push({ target: plugin.target, label: plugin.label });
+      } catch (e) {
+        console.error(e.message);
+      }
+    });
+
+    shell.exec(
+      `rsync -av "${path.resolve("./plugins")}" "${Config.build_path(
+        exp_info.src
+      )}" --exclude=.git`
+    );
+    return plugins;
+  }
+
+  static preProcessPageScopePlugins(options) {
+    const pluginConfigFile = Plugin.getConfigFileName(options.env);
+    const pluginConfig = require(pluginConfigFile);
+    const plugins = {};
+    pluginConfig
+      .filter((p) => p.scope === PluginScope.PAGE)
+      .forEach((element) => {
+        plugins[element.id] = element;
+      });
+
+    // console.log(plugins["plugin-bug-report"].attributes.bug_options);
+    return plugins;
+  }
+  static processPageScopePlugins(page, options) {
+    const pluginConfigFile = Plugin.getConfigFileName(options.env);
+    const pluginConfig = require(pluginConfigFile);
+
+    const pageScopePlugins = pluginConfig.filter(
+      (p) => p.scope === PluginScope.PAGE
+    );
+
+    const html = fs.readFileSync(path.resolve(page.targetPath()));
+    const dom = new JSDOM(html);
+    const { document } = dom.window;
+
+    pageScopePlugins.forEach((plugin) => {
+      // Render the Plugin UI component inside the parent
+      const pluginParent = document.getElementById(plugin.id);
+      if (pluginParent) {
+        // Write code to process a template file and add to the parent
+      }
+
+      // add the css-modules in the head
+      plugin.css_modules &&
+        plugin.css_modules.forEach((css) => {
+          const cssNode = document.createElement("link");
+          cssNode.rel = "stylesheet";
+          cssNode.href = css;
+
+          document.head.appendChild(cssNode);
+        });
+
+      // add the js-modules at the bottom of the body
+      plugin.js_modules &&
+        plugin.js_modules.forEach((mjs) => {
+          const scriptNode = document.createElement("script");
+          scriptNode.type = "module";
+          scriptNode.src = mjs;
+
+          document.body.appendChild(scriptNode);
+        });
+    });
+    fs.writeFileSync(page.targetPath(), dom.serialize());
+  }
+
+  static processPostBuildPlugins(exp_info, options){
+    const pluginConfig = require(Plugin.getConfigFileName(options.env));
+  
+      const postBuildScopePlugins = pluginConfig.filter(
+        (p) => p.scope === PluginScope.POSTBUILD
+      );
+      if(!fs.existsSync('plugins'))
+      {
+        shell.exec('mkdir plugins');
+      }
+  
+      postBuildScopePlugins.forEach((plugin) => {
+        try {

```

Comment:
Lines 178-190 can probably become a separate function named as prepareRepo or updateRepo. I assume other plugins also need this functionality